### ISSUE TITLE
Upload array of mat4 uniforms?

### ISSUE THREAD

**Issue by kungfooman** [Tomius/oglwrap#5]
Hey Tomius, currently I wonder how to upload an array of matrices via oglwrap. I searched through the code and I only found this, with count parameter forced to 1:

https://github.com/Tomius/oglwrap/blob/master/uniform.h#L518

	#if OGLWRAP_DEFINE_EVERYTHING || defined(glUniform4dv)
	template<>
	inline void UniformObject<glm::dvec4>::set(const glm::dvec4& vec) {
	  glUniform4dv(location_, 1, glm::value_ptr(vec));
	}

Unfortunately I never learned to properly do template programming and I wonder how this could be implemented.

E.g. when I have 7 bindposes for skeletal animation:

	gl::Uniform<glm::mat4[7]> u_bindposes(proggy, "u_bindposes");

Could a template extract the array length?

Or maybe like, with a default value of 1:

	gl::Uniform<glm::mat4, 7> u_bindposes(proggy, "u_bindposes");

Or probably something like this, to dynamically set the number of elements:

	glm::mat4 matrices[7];
	gl::Uniform<glm::mat4> u_bindposes(proggy, "u_bindposes");
	u_bindposes.Set(matrices, 7);

Or I'm just missing something, so this might be already possible without falling back to legacy OpenGL. ^^

**Comment by Tomius** [comment]
There is a class called IndexedUniform to upload array uniform data, but it has to be used for each element of the array:
```c++
glm::mat4 matrices[7];
for (int i = 0; i < 7; ++i) {
  gl::IndexedUniform<glm::mat4> u_bindposes(proggy, "u_bindposes", i);
  u_bindposes.Set(matrices[i]);
}
```

Also, if you use LasyUniforms (that queries the uniform position in the first use, not in the constructor), you can simply index the LasyUniform object. Here's a code about a skeletal animation I wrote a few years ago:
```c++
void AnimatedMeshRenderer::uploadBoneInfo(gl::LazyUniform<glm::mat4>& bones) {
  for (unsigned i = 0; i < skinning_data_.num_bones; i++) {
      bones[i] = skinning_data_.bone_info[i].final_transform;
  }
}

```

 I will make uploading an array of uniforms at once possible soon, as it seems a lot more efficient.

**Comment by Tomius** [comment]
Fixed in f0a3f70d2f63c8d391a6d8eb8f110f1f3c7045b7, 2a441ca833d2612b16e8dc8f407acf1e4d6fb618.

Now you can use it like this:
```c++
glm::mat4 matrices[7];
gl::LazyUniform<glm::mat4> u_bindposes(proggy, "u_bindposes");
u_bindposes = matrices; // uploads 7 of glm::mat4s (works for gl::Uniform too)

// also works with other containers, not just array
std::vector<glm::mat> mats (7);
u_bindposes = mats;

// the 5th matrix has changed, only want to reupload that
u_bindposes[5] = glm::mat4(...); // requires gl::LazyUniform
```
